tests/python_api/external_source_reader.py

```diff
@@ -5,16 +5,18 @@
 import amd.rocal.types as types
 import os
 import numpy as np
-import cupy as cp
 import cv2
-from PIL import Image
+import sys
 
 
 def main():
-    batch_size = 3
+    if  len(sys.argv) < 3:
+        print ('Please pass cpu/gpu batch_size')
+        exit(0)
+    batch_size = int(sys.argv[2])
+    device = "cpu" if sys.argv[1] == "cpu" else "gpu"
     data_dir = os.environ["ROCAL_DATA_PATH"] + \
         "rocal_data/coco/coco_10_img/train_10images_2017/"
```

Comment:
Not sure how the app is working for you @swetha097 This path does not exist. https://github.com/ROCm/MIVisionX-data/tree/main/rocal_data/coco/coco_10_img/images
Please update and retest

Reply:
Updated the path.